sdk/core/src/main/java/com/klaviyo/coresdk/Klaviyo.kt

```diff
@@ -1,73 +1,140 @@
+     *
+     * This should be called when you collect additional data about your user
+     * (e.g. first and last name, or an address)
+     *
+     * @param propertyKey
+     * @param value
+     * @return
+     */
+    fun setProfileAttribute(propertyKey: KlaviyoProfileAttributeKey, value: String): Klaviyo =
+        setProfile(Profile().also { it[propertyKey] = value })
+
+    /**
+     * Queues a request to identify profile properties to the Klaviyo API
+     *
+     * Any new identifying properties (externalId, email, phone) will be saved to the current
+     * internally tracked profile. Otherwise any identifiers missing from [profile] will be
+     * populated from the current internally tracked profile info.
+     *
      * Identify requests track specific properties about a user without triggering an event
      *
-     * @param properties A map of properties that define the user
+     * @param profile A map of properties that define the user
+     * @return
      */
-    fun identify(properties: KlaviyoCustomerProperties) {
-        val request = IdentifyRequest(KlaviyoConfig.apiKey, properties)
-        processRequest(request)
+    fun setProfile(profile: Profile): Klaviyo = apply {
+        // TODO debounce so fluent setters don't have to create 1 request per call
+        // Note - is there value to tighter coupling the UserInfo update with the API call
+        UserInfo.mergeProfile(profile)
+        Registry.apiClient.enqueueProfile(profile)
     }
 
     /**
-     * Processes the given [KlaviyoRequest] depending on the SDK's configuration.
-     * If the batch queue is enabled then requests will be batched and sent in groups.
-     * Otherwise the request will send instantly.
-     * These requests are sent to the Klaviyo asynchronous APIs
+     * Clears all stored profile identifiers (e.g. email or phone)
+     *
+     * This should be called whenever an active user in your app is removed
+     * (e.g. after a logout)
+     *
+     * @return
+     */
+    fun resetProfile(): Klaviyo = apply {
+        UserInfo.reset()
+        Registry.apiClient.enqueueProfile(UserInfo.getAsProfile())
+    }
+
+    /**
+     * Queues a request to track an [Event] to the Klaviyo API
+     * The event will be associated with the profile specified by the [Profile]
+     * If profile is not set, this will fallback on the current profile identifiers
+     *
+     * @param event Name of the event metric
+     * @param properties Additional properties associated to the event that are not for identifying the profile
+     * @return
      */
-    private fun processRequest(request: KlaviyoRequest) {
-        if (KlaviyoConfig.networkUseAnalyticsBatchQueue) {
-            request.batch()
-        } else {
-            request.process()
-        }
+    fun createEvent(event: KlaviyoEventType, properties: Event? = null): Klaviyo = apply {
+        Registry.apiClient.enqueueEvent(
+            event,
+            properties ?: Event(),
```

Comment:
should this default just be in the params instead? `, properties: Event = Event()`

Reply:
Yup.